### PR TITLE
Make Controller::disconnect() wait for server to close properly

### PR DESCRIPTION
**Pull request by pdamasceno**

### CODE REVIEW

File: obs-studio-client/source/controller.cpp [hunk 1]
```diff
@@ -316,6 +316,12 @@ void Controller::disconnect()
 {
 	if (m_isServer) {
 		m_connection->call_synchronous_helper("System", "Shutdown", {});
+
+		// Waiting for server to close
+		while (is_process_alive(procId)) {
```
Comment (EddyGharbi) [inline obs-studio-client/source/controller.cpp:321]: We cannot do that here. This is exposing to a potential infinite loop that would freeze the app entirely.

**Comment by EddyGharbi** [comment]
Let's put the PR on hold while we figure out if we actually needs it.

**Comment by pdamasceno** [comment]
This is not needed. Found out the reason of the lost IPC connection when starting a OBS process. We can cancel this PR.

**Comment by EddyGharbi** [comment]
Closing it 👍

**Pull request closed without merging**